sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/action/ActionReflectiveRouter.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl.action
+
+import java.lang.reflect.Method
+
+import akka.NotUsed
+import akka.stream.javadsl.Source
+import kalix.javasdk.action.Action
+import kalix.javasdk.action.MessageEnvelope
+import kalix.javasdk.impl.action.ActionRouter
+
+class ActionReflectiveRouter[A <: Action](override val action: A) extends ActionRouter[A](action) {
+
+  private val allHandlers: Map[String, Method] =
+    // FIXME: names must be unique, overloading shouldn't be allowed,
+    //  we should detect it here and fail-fast
+    action.getClass.getDeclaredMethods.toList
+      // handlers are all methods returning Effect
+      .filter(_.getReturnType == classOf[Action.Effect[_]])
+      // and with one single input param
+      .filter(_.getParameters.length == 1)
+      .map { javaMethod => (javaMethod.getName.capitalize, javaMethod) }
+      .toMap
```

Comment:
This is much more complicated than expected. 

The direction that we are taking is to take a method defined in Java and generate something completely different in proto. 


For example....

```java
@PostMapping("/foo/{bar}")
public Effect<...> foo(@PathVariable("bar") String bar, Foo foo)
```

We will need to generate a FooRequest that will look like:

```proto
message FooRequest {
  string bar = 1;
  Any payload = 2;
}
```

and a gRPC method as: 

```proto
rpc Foo(FooRequest) returns (Any)
```

In the router, we will need to have the knowledge that a command called `Foo`, will come with a payload of `FooRequest` and that we need to open it up, take all the parts and then call `JavaClass.foo(String, Foo)`.

That said, we are here rebuilding teh `allHandlers` map, but that's something that build somewhere as pass a param here. And it will be more than just a map from `String -> Method`.